modules/sfp_flickr.py

```diff
@@ -0,0 +1,174 @@
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+import json
+import re
+import time
+from sflib import SpiderFoot, SpiderFootPlugin, SpiderFootEvent
+
+
+class sfp_flickr(SpiderFootPlugin):
+    """Flickr:Footprint,Investigate,Passive:Social Media::Look up e-mail addresses on Flickr."""
+
+    # Default options
+    opts = {
+        'pause': 1,      # number of seconds to pause between fetches
+        'per_page': 200, # max number of results per page
+        'maxpages': 20   # max number of pages to fetch
+    }
+
+    # Option descriptions
+    optdescs = {
+        'pause': "Number of seconds to pause between fetches.",                                                                                                                                                    
+        'per_page': "Maximum number of results per page.",
+        'maxpages': "Maximum number of pages of results to fetch."
+    }
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.__dataSource__ = "Flickr"
+        self.results = list()
+
+        for opt in userOpts.keys():
+            self.opts[opt] = userOpts[opt]
+
+    # What events is this module interested in for input
+    def watchedEvents(self):
+        return ["DOMAIN_NAME"]
+
+    # What events this module produces
+    def producedEvents(self):
+        return ["EMAILADDR"]
+
+    # Retrieve API key
+    def retrieveApiKey(self):
+        res = self.sf.fetchUrl("https://www.flickr.com/", timeout=self.opts['_fetchtimeout'], useragent=self.opts['_useragent'])
+
+        if res['content'] is None:
+            return None
+
+        keys = re.findall(r'YUI_config.flickr.api.site_key = "([a-zA-Z0-9]+)"', res['content'])
+
+        if not keys:
+            return None
+
+        return keys[0]
+
+    # Query the REST API
+    def query(self, qry, api_key, page=1, per_page=200):
+        output = "json"
+        url = "https://api.flickr.com/services/rest?"
+        url += "sort=relevance&parse_tags=1&content_type=7&extras=description,owner_name,path_alias,realname&"
+        url += "hermes=1&hermesClient=1&reqId=&nojsoncallback=1&viewerNSID=&method=flickr.photos.search&csrf=&lang=en-US&"
+        url += "per_page=" + str(per_page) + "&page=" + str(page) + "&text=" + qry + "&api_key=" + api_key + "&format=" + output
+
+        res = self.sf.fetchUrl(url, timeout=self.opts['_fetchtimeout'], useragent=self.opts['_useragent'])
+
+        return res['content']
+
+    # Handle events sent to this module
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+
+        if eventData in self.results:
+            return None
+        else:
+            self.results.append(eventData)
+
+        self.sf.debug("Received event, " + eventName + ", from " + srcModuleName)
+
+        if not eventName == "DOMAIN_NAME":
+            return None
```

Comment:
Not really necessary, because the module will only receive events it has expressed interest in from watchedEvents().